PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java

```diff
@@ -139,15 +131,121 @@ public void actionPerformed(ActionEvent e) {
         ButtonGroup buttonGroup = new ButtonGroup();
         buttonGroup.add(deviceLoginRadioButton);
         buttonGroup.add(automatedRadioButton);
+        buttonGroup.add(azureCliRadioButton);
         deviceLoginRadioButton.setSelected(true);
-        refreshAuthControlElements();
 
         init();
     }
 
+    public AuthMethodDetails getAuthMethodDetails() {
+        return authMethodDetailsResult;
+    }
+
+    public static SignInWindow go(AuthMethodDetails authMethodDetails, Project project) {
+        SignInWindow d = new SignInWindow(authMethodDetails, project);
+        d.show();
+        if (d.getExitCode() == DialogWrapper.OK_EXIT_CODE) {
+            return d;
+        }
+
+        return null;
+    }
+
+    @Override
+    public void doCancelAction() {
+        authMethodDetailsResult = authMethodDetails;
+        super.doCancelAction();
+    }
+
+    @Override
+    public void doHelpAction() {
+        JXHyperlink helpLink = new JXHyperlink();
+        helpLink.setURI(URI.create("https://docs.microsoft.com/en-us/azure/azure-toolkit-for-intellij-sign-in-instructions"));
+        helpLink.doClick();
+    }
+
+    @Nullable
+    @Override
+    protected String getDimensionServiceKey() {
+        return "SignInWindow";
+    }
+
+    @Nullable
+    @Override
+    protected JComponent createCenterPanel() {
+        return contentPane;
+    }
+
+    @Override
+    protected void doOKAction() {
+        authMethodDetailsResult = new AuthMethodDetails();
+        if (automatedRadioButton.isSelected()) { // automated
```

Comment:
How about to use a `doLogin()` interface for every auth method with the special logic code inside?

Reply:
Yes it could be better to abstract the interface for auth, this may be done in later milestone